solutions/devsprint-cecilia-saraiva-2/FinanceAppTests/Service/FinanceServiceTests.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  FinanceServiceTests.swift
+//  FinanceAppTests
+//
+//  Created by Vitor Conceicao on 16/08/22.
+//
+
+import XCTest
+@testable import FinanceApp
+
+final class FinanceServiceTests: XCTestCase {
+    
+    func test_fetchHomeData_shouldnCallWithCorrectURLWhenCalled() {
+        let (sut, networkClient) = makeSut()
+        let stringURL = "https://raw.githubusercontent.com/devpass-tech/challenge-finance-app/main/api/home_endpoint.json"
+        
+        let expectation = XCTestExpectation(description: "Wait request")
+        
+        sut.fetchHomeData { _ in
+            XCTAssertTrue(networkClient.performRequestCalled)
+            XCTAssertEqual(networkClient.calledMethods, [.performRequest(url: stringURL)])
```

Comment:
[sugestao] podemos fazer as asserções fora do completion handler e simplesmente armazenar em variáveis locais o que for necessário pra asserções (eg data).

Reply:
Mas ai o teste não falharia? Pois eu preciso esperar a closure ser executada e o expect resolve isso pra mim.